projects/packages/jetpack-mu-wpcom/src/features/calypso-locale-sync/calypso-locale-sync.php

```diff
@@ -0,0 +1,34 @@
+<?php
+/**
+ * Sync Calypso locale when wp-admin locale is updated via /wp-admin/profile.php.
+ *
+ * @package automattic/jetpack-mu-wpcom
+ */
+
+use Automattic\Jetpack\Connection\Client;
+
+if ( function_exists( 'wpcom_is_nav_redesign_enabled' ) && wpcom_is_nav_redesign_enabled() ) {
+	add_filter(
+		'profile_update',
+		function ( $user_id, $old_user_data, $user_data ) {
+			$locale = $user_data['locale'];
+
+			if ( ! $locale ) {
+				// No locale means the "Site Default" option which is the Site language (WPLANG).
+				$locale = get_option( 'WPLANG', '' );
+			}
+
+			Client::wpcom_json_api_request_as_user(
```

Comment:
I believe we should only call this when `$locale` is different from `$old_user_data`'s locale. Otherwise we will be calling this endpoint every time the user changes anything in their profile.

Reply:
Sounds good 👍
I will also change the endpoint name as you have suggested.

Reply:
I had to change the hook to [insert_user_meta](https://developer.wordpress.org/reference/hooks/insert_user_meta) because [profile_update](https://developer.wordpress.org/reference/hooks/profile_update/) fires after the update.